aws-serverless-java-container-springboot3/src/main/java/com/amazonaws/serverless/proxy/spring/AWSHttpUtils.java

```diff
@@ -0,0 +1,183 @@
+package com.amazonaws.serverless.proxy.spring;
+
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.cloud.function.serverless.web.ServerlessHttpServletRequest;
+import org.springframework.cloud.function.serverless.web.ServerlessMVC;
+import org.springframework.util.FileCopyUtils;
+import org.springframework.util.MultiValueMapAdapter;
+import org.springframework.util.StringUtils;
+
+import com.amazonaws.serverless.proxy.AwsHttpApiV2SecurityContextWriter;
+import com.amazonaws.serverless.proxy.AwsProxySecurityContextWriter;
+import com.amazonaws.serverless.proxy.RequestReader;
+import com.amazonaws.serverless.proxy.SecurityContextWriter;
+import com.amazonaws.serverless.proxy.internal.servlet.AwsHttpServletResponse;
+import com.amazonaws.serverless.proxy.internal.servlet.AwsProxyHttpServletResponseWriter;
+import com.amazonaws.serverless.proxy.model.AwsProxyRequest;
+import com.amazonaws.serverless.proxy.model.AwsProxyResponse;
+import com.amazonaws.serverless.proxy.model.HttpApiV2ProxyRequest;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import jakarta.servlet.ServletContext;
+import jakarta.servlet.http.HttpServletRequest;
+
+public class AWSHttpUtils {
```

Comment:
Naming things is the hardest thing to do but we should replace the generic name with a more specific one. If extractVersion is moved out, maybe HttpTransformationUtils, HttpProcessingUtils or similar?!
Let's also add a private constructor to prevent instantiation.

Reply:
I'll keep it for now as it matches Spring idioms (e.g., `StringUtils`, `CollectionUtils` basically `*Utils`) and it allows it to be a bit more generic where one can keep reusable operations for specific concept such as this case where we are providing  a set of reusable operations for the entire concept of serving SpringMVC web requests via AWS Lambda. 
In the future when things become a bit more clear we can refactor it more, but for now to ensure no one outside of this package can use it i made it package protected.

Reply:
Ok, I've changed the name to `AwsSpringHttpProcessingUtils` to make it less generic